### PR TITLE
Add tutorial

### PR DESCRIPTION
**Pull request by jcp19**

### CODE REVIEW

File: docs/tutorial-examples/channels.gobra [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+requires (*pc).SendGotPerm() == PredTrue!<!>;
+requires (*pc).RecvGivenPerm() == PredTrue!<!>;
+requires (*pc).RecvGotPerm() == sendInvariant!<_!>;
+func inc(pc *chan *int, ghost x int) {
+    fold PredTrue!<!>()
+    res, ok := <- *pc
+    if (ok) {
+        unfold sendInvariant!<_!>(res)
+        // we should have write access and thus can write to it
+        // before being able to fold again, we have to revert the value:
```
Comment (ArquintL) [inline docs/tutorial-examples/channels.gobra:46]:
```suggestion
        // we should have write access and thus can write to it:
```

File: docs/tutorial-examples/interface.gobra [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+type stream interface {
+		pred mem()
+
+		requires acc(mem(), 1/2)
+		pure hasNext() bool
+
+		requires mem() && hasNext()
+		ensures  mem()
+		next() interface{}
```
Comment (ArquintL) [inline docs/tutorial-examples/interface.gobra:15]:
```suggestion
	pred mem()

	requires acc(mem(), 1/2)
	pure hasNext() bool

	requires mem() && hasNext()
	ensures  mem()
	next() interface{}
```

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,693 @@
+# A Tutorial on Gobra
+## Introduction
+*Gobra* is an automated, modular verifier for heap-manipulating, concurrent Go programs. 
+It supports a large subset of Go, including Go’s interfaces and primitive data structures.
+Gobra verifies memory safety, crash safety, data-race freedom, and partial 
+correctness based on user-provided specifications.
+Gobra takes as input a `.gobra` file containing a Go program annotated with 
+specifications such as function pre- and postconditions and loop invariants.
+
+In order to prove the correctness of a program, Gobra verifies
```
Comment (ArquintL) [inline docs/tutorial.md:10]:
```suggestion
In order to prove correctness of a program, Gobra verifies
```

File: docs/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,693 @@
...
+all functions and methods according to their specifications. If one of the
+methods fails to meet its specification, the verification fails and Gobra 
+reports the error.
```
Comment (ArquintL) [inline docs/tutorial.md:13]:
```suggestion
methods fails to meet its specification, verification fails and Gobra 
reports the error.
```

File: docs/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,693 @@
...
+
+This tutorial provides a practical introduction to Gobra and showcases some of its main features. 
+
+TODO: OUTLINE: Brief introduction: Say what this section shows: “First, we will introduce Gobra’s top-level declarations. This tutorial will provide more details for their use in the subsequent sections”
```
Comment (ArquintL) [inline docs/tutorial.md:17]: I don't really see a need as the next subsection already contains an outline

File: docs/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,693 @@
...
+
+### Package Clauses
+
+```go
+package pkgName
+```
+
+- The `package` keyword defines the package to which the file belongs.
+- Every Gobra file belongs necessarily to one and only one package.
+
```
Comment (ArquintL) [inline docs/tutorial.md:33]:
```suggestion
- Files belonging to the same package have to be located in the same directory.
- A directory can only contain files belonging to a single package.

```

File: docs/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,693 @@
...
+```
+- create a new type with the same underlying type and operations as the provided type.
+- The two identifiers correspond to different types. 
+- In this example, `TypeDef` is a new type whose underlying type is `SomeType`. 
+
+
+#### Global Constants
+```go
+const untypedConst = constExpr // untyped constant
+const typedConst type = consExpr // constExpr must be known at compile-time
```
Comment (ArquintL) [inline docs/tutorial.md:112]:
```suggestion
const typedConst type = constExpr // constExpr must be known at compile-time
```

**Comment by Felalolf** [comment]
Currently, we are working on the tutorial here https://hackmd.io/gXb5okc0Rxiu4a3Wz4Q0Ag?both#Data-Types
I will commit this verison and start a PR when it is done.

File: src/test/resources/regressions/examples/tutorial-examples/multi-chanel.gobra [hunk 8]
```diff
@@ -0,0 +1,103 @@
+// Any copyright is dedicated to the Public Domain.
```
Comment (ArquintL) [inline src/test/resources/regressions/examples/tutorial-examples/multi-chanel.gobra:1]: spelling mistake in the filename, should be `multi-channel.gobra`

File: src/test/resources/regressions/examples/tutorial-examples/permissions.gobra [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+func client1() {
+	p := &pair{1,2}
+	p.sumPair(42)
+	assert p.left == 43
+}
+
+func client2() {
+	x@ := pair{1,2} // if the reference of an address is taken, then add @
```
Comment (ArquintL) [inline src/test/resources/regressions/examples/tutorial-examples/permissions.gobra:30]: maybe better:
```suggestion
	x@ := pair{1,2} // if taking the reference of a variable should be possible, then add @
```

**Review by ArquintL — APPROVED** [review]
I skipped the tutorial and looked only at the examples. I've left a few minor comments but overall it looks great!

**Pull request merged**